File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const res = [];
        const map = {};
        for (let num of nums){
            if(!map[num])
                map[num] = 1;
            else
                map[num]++;
        }

        const freq = Array.from({length: nums.length + 1}, () => []);

        for (const m in map){
            freq[map[m]].push(m);
        }

        for(let i = freq.length - 1; i>=0; i--){
            for (let j = 0; j < freq[i].length; j++){
                res.push(freq[i][j]);
                if(res.length === k)
                    return res;
            }
        }
    }
}

File: Data Structures & Algorithms/binary-search/submission-0.cs
public class Solution {
    public int Search(int[] nums, int target) {
        int l = 0; int r = nums.Length - 1;

        return BinarySearch(l,r,nums,target);
    }

    public int BinarySearch(int l, int r, int[] nums, int target){
        if(l > r) return -1;
        int mid = l + ((r-l)/2);
        
        if(nums[mid] == target) return mid;

        int res = nums[mid] > target 
            ? BinarySearch(l, mid - 1, nums, target) : BinarySearch(mid + 1, r, nums, target);

        return res;
    }
}
